Sprinkle of Yeezus/Settings.swift

```diff
@@ -11,52 +11,68 @@ import UIKit
     
-    @IBAction func NotificationSwitch(_ sender: UISwitch) {
-        if Switch.isOn == true{
-            UNUserNotificationCenter.current().requestAuthorization(options: [.alert, .sound], completionHandler: {didAllow, error in})
-            
-            DescLabel.isHidden = false
-            DescButton.isHidden = false
-            
-            notificationsOn = true
-        } else {
-            DescLabel.isHidden = true
-            DescButton.isHidden = true
-            notificationsOn = false
-        }
+    // MARK: - Lifecycle
+    override func viewDidLoad() {
+        sprinkleNotifications(sprinkleList)
     }
     
-    @IBOutlet weak var TimePicked: UIDatePicker!
-    var sprinkleTimePicked = Date()
-    @IBAction func TimePicker(_ sender: UIDatePicker){
-        sprinkleTimePicked = TimePicked.date
+    // MARK: - IBAction
+    @IBAction private func NotificationSwitch(_ sender: UISwitch) {
+        if sender.isOn {
+            notificationCenter.requestAuthorization(options: [.alert, .sound], completionHandler: { didAllow, error in
+                DispatchQueue.main.async {
+                    sender.isOn = didAllow
+                    self.updateUI()
+                }
+            })
+        }
+        
+        updateUI()
     }
     
-    func sprinkleNotifications(_ sprinkleList: Array<Sprinkle>) {
-        if notificationsOn == true{
-            
-            let notification = UNMutableNotificationContent()
-            let notificationText = pickRandomQuote(sprinkleList)
-            notification.body = " \"\(notificationText.quote)\" \(notificationText.quoteSource), \(notificationText.date)"
-            
-            let timeForSprinkle = TimePicked.date
-            let components = Calendar.current.dateComponents([.hour, .minute], from: timeForSprinkle)
-            InfoLabel.text = "Sprinkles will be sent every day at \(components.hour):\(components.minute)"
-            let notificationTime = UNCalendarNotificationTrigger(dateMatching: components, repeats: true)
-            
-        }
+    @IBAction private func TimePicker(_ sender: UIDatePicker){
+        sprinkleTimePicked = sender.date
+    }  
+    
+    // MARK: - Private
+    private func updateUI() {
+        [DescLabel, DescButton].forEach { $0.isHidden = !Switch.isOn }
     }
     
-    override func viewDidLoad() {
-        sprinkleNotifications(sprinkleList)
+    private func sprinkleNotifications(_ sprinkleList: Array<Sprinkle>) {
+        notificationCenter.removeAllPendingNotificationRequests()
+        
+        guard Switch.isOn else {
+            return
+        }
+        
+        let notification = UNMutableNotificationContent()
+        let notificationText = pickRandomQuote(sprinkleList)
+        notification.body = "\"\(notificationText.quote)\" \(notificationText.quoteSource), \(notificationText.date)"
+
+        let components = Calendar.current.dateComponents([.hour, .minute], from: sprinkleTimePicked)
```

Comment:
This code is relatively the same, I just gave the `hour` and `minute` default values because there is a warning when you use optional values in string interpolation.